mkosi/resources/__init__.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: PSF-2.0
+# Based on code from https://github.com/python/cpython/blob/main/Lib/importlib/resources/_common.py
+
+import contextlib
+import functools
+import os
+import stat
+import tempfile
+from collections.abc import Iterator
+from contextlib import AbstractContextManager
+from importlib.abc import Traversable
+from pathlib import Path
+
+
+def make_executable_if_script(path: Path) -> Path:
+    with path.open() as f:
+        buf = f.read(3)
+        if buf == "#!/":
```

Comment:
This should fix the test failure.
```suggestion
    with path.open("rb") as f:
        buf = f.read(3)
        if buf == b"#!/":
```